### PR TITLE
Add assert to make sure getOccurences at position only returns results f...

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Add assert to make sure getOccurences at position only returns results for the file we request the occurences for.

This should help figure out why #1153 is happening.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -3571,8 +3571,24 @@ module ts {
...
+                    let targetFile = getCanonicalFileName(normalizeSlashes(value.fileName));
+                    Debug.assert(sourceFile == targetFile, `Unexpected file in results. Found results in ${targetFile} expected only results in ${sourceFile}.`);
+                });
+            }
+
+            return results;
+        }
+
+        /// References and Occurrences
+        function getOccurrencesAtPositionCore(fileName: string, position: number): ReferenceEntry[] {
```
Comment (mhegazy) [inline src/services/services.ts]: can we just filter instead. this is going to fail anyways. just remove the entries.

Comment (paulvanbrenk) [inline src/services/services.ts]: This won't crash VS, and I rather have this fixed in the script side to prevent we do unnecessary work there.

Comment (mhegazy) [inline src/services/services.ts]: i meant to filter on the script side. you know the caller asked for a specific file, so just filter results that are not there.

Comment (CyrusNajmabadi) [inline src/services/services.ts]: I think the idea is that we shouldn't be finding more results anyways.  It represents a bug in our code that we should not be trying to workaround.  Instead, we should track down the actual bug and fix it.

**Pull request merged**